x-pack/plugins/cases/server/client/metrics/alert_details.ts

```diff
@@ -5,33 +5,118 @@
  * 2.0.
  */
 
-import { CaseMetricsResponse } from '../../../common/api';
+import { merge } from 'lodash';
+
+import { AlertHostsMetrics, AlertUsersMetrics, CaseMetricsResponse } from '../../../common/api';
+import { createCaseError } from '../../common/error';
+import { AggregationFields, FrequencyResult, UniqueCountResult } from '../../services/alerts/types';
+import { CasesClient } from '../client';
+import { CasesClientArgs } from '../types';
 import { MetricsHandler } from './types';
 
 export class AlertDetails implements MetricsHandler {
-  /**
-   * This boolean protects against the metrics being queried multiple times. The applyMetrics function could be called
-   * once for each feature. All the metrics will be retrieved when the first applyMetrics is called though.
-   */
-  private retrievedMetrics: boolean = false;
+  private requestedFeatures: Set<AggregationFields> = new Set();
```

Comment:
this `requestedFeatures` is a little bit confusing to me. The name suggests it contains features (strings) but It stores aggregation fields, maybe we could name it `featureAggregationFields` or something more related?

Another question is why this is a `Set`? it seems we always do `Array.from()` when we read it, so why not use an Array directly? Do we have any convention in favor of `Set` instead of `Array`? just curious.

Reply:
> The name suggests it contains metrics feature names, but It stores aggregation fields which are the es index field names, maybe we could name it featureAggregationFields or something more related?

Yeah good point, I'll change that.

> Do we have any convention in favor of Set instead of Array? just curious.

I try to prefer a Set when there's the possibility for duplicate entries. It's probably overkill here since it'd be a programatic error if we had duplicates.